src/app/components/core/control-dispatcher/pipelines/add-pipeline/add-control-pipeline.component.ts

```diff
@@ -0,0 +1,682 @@
+      } else {
+        this.destinationTypeList = data;
+      }
+    }, error => {
+      if (error.status === 0) {
+        console.log('service down ', error);
+      } else {
+        this.alertService.error(error.statusText);
+      }
+    });
+  }
+
+  /**
+  * To upload script files of a configuration property
+  * @param categoryName name of the configuration category
+  * @param files : Scripts array to uplaod
+  */
+  public uploadScript(categoryName: string, files: any[]) {
+    this.fileUploaderService.uploadConfigurationScript(categoryName, files);
+  }
+
+  /**
+  * Get scripts to upload from a configuration item
+  * @param configuration  edited configuration from show configuration page
+  * @returns script files to upload
+  */
+   getScriptFilesToUpload(configuration: any) {
+    return this.fileUploaderService.getConfigurationPropertyFiles(configuration);
+  }
+
+  /**
+   * update plugin configuration
+   */
+   updateConfiguration(categoryName: string, configuration: any, type: string) {
+    const files = this.getScriptFilesToUpload(configuration);
+    if (files.length > 0) {
+      this.uploadScript(categoryName, files);
+    }
+
+    if (isEmpty(configuration)) {
+      return;
+    }
+
+    this.apiCallsStack.push(this.configService.
+      updateBulkConfiguration(categoryName, configuration)
+      .pipe(map(() => ({ type, success: true })))
+      .pipe(catchError(e => of({ error: e, failed: true }))));
+  }
+
+  onSubmit(form: NgForm) {
+    const formData = cloneDeep(form.value);
+    let { name } = formData;
+    const payload = {
+      name: name.trim(),
+      execution: this.selectedExecution,
+      source: {"type": this.selectedSourceType.cpsid, "name": this.selectedSourceName},
+      destination: {"type": this.selectedDestinationType.cpdid, "name": this.selectedDestinationName},
+      filters: this.filterPipeline ? this.filterPipeline : [],
+      enabled: this.isPipelineEnabled
+    }
+    if (!isEmpty(this.changedFilterConfig) && this.filterConfigurationCopy?.key) {
+      this.updateConfiguration(this.filterConfigurationCopy.key, this.changedFilterConfig, 'filter-config');
+    }
+
+    if (this.apiCallsStack.length > 0) {
+      this.ngProgress.start();
+      forkJoin(this.apiCallsStack).subscribe((result) => {
+        result.forEach((r: any) => {
+          this.ngProgress.done();
+          if (r.failed) {
+            if (r.error.status === 0) {
+              console.log('service down ', r.error);
+            } else {
+              this.alertService.error(r.error.statusText);
+            }
+          } else {
+            this.response.handleResponseMessage(r.type);
+          }
+        });
+        form.reset();
+        this.apiCallsStack = [];
+      });
+    }
+
+    // if ((payload.source.type !== 1 || payload.source.type !== 3 && payload.source.name === '') || (payload.destination.type !== 4 && payload.destination.name === '')) {
+    //   return;
+    // }
```

Comment:
remove?

Reply:
I'll use this condition in the patch PR, so I kept it.